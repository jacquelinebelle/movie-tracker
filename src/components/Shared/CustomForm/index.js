import React from 'react';
import { Redirect } from 'react-router'; 
import './CustomForm.scss';

export const CustomForm = ({ title, formFields, onSubmit, isLoggedIn, error }) => {
    

    const fields = formFields.map(field => {
        return(
            <>
                <label className={`${field.type}-input`}>{field.type}</label>
                <input type="text" placeholder={field.type}
                    name={field.type}
                    value={field.value}
                    onChange={field.onChange} />
            </>
        )
    }) 

    return( 
        <>
            <h2>{title}</h2>
            <form>
                {fields}
                <button onClick={onSubmit}>Submit</button>
                {isLoggedIn && <Redirect to="/" />}
            </form>
            {error.isError && <h2>{error.message}</h2>}
        </>
    )
}

export default CustomForm;