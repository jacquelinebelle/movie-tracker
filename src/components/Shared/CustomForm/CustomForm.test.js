import React from 'react';
import CustomForm from './index.js';
import { shallow } from 'enzyme';


describe('CustomForm', () => {
  let wrapper;
  let formFields;
  let error;

  beforeEach(() => {
    formFields = ['a', 'b', 'c']
    error = {
      message: 'error'
    }
    wrapper = shallow(<CustomForm formFields={formFields} error={error}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})