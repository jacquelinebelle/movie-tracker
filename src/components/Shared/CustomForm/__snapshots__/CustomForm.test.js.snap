// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomForm should match the snapshot 1`] = `ShallowWrapper {}`;
